x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts

```diff
@@ -268,6 +268,7 @@ export const MachineLearningSpecificRuleParams = z.object({
   type: z.literal('machine_learning'),
   anomalyThreshold: AnomalyThreshold,
   machineLearningJobId: z.array(z.string()),
+  alertSuppression: AlertSuppressionCamel.optional(),
```

Comment:
This is adding a new optional param to the rule, and so may need an ["intermediate release"](https://docs.google.com/document/d/1mU5jlIfCKyXdDPtEzAz1xTpFXFCWxqdO5ldYRVO_hgM/edit?usp=sharing) to be able to handle rolling back a Kibana version - what happens if a rule with this param is rolled back to the previous version that doesn't use it?  

It wasn't clear from the PR description whether this has been taken into account.  

Looks like that's the only ResponseOps change - the snapshot of the rule params showing a change here.

Reply:
@pmuellr thanks for calling this out. I think the same applies to [](), no? It looks like the reason that one wasn't caught is because the `siem.esqlRule` is not accounted for in [this index](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/integration_tests/serverless_upgrade_and_rollback_checks.test.ts#L56-L71)

Reply:
@pmuellr so in both this PR and the analogous [ES|QL Suppression PR](https://github.com/elastic/kibana/pull/180927), depending on the feature flag, we:

* show/hide the corresponding form fields, 
* [accept or strip out those fields on the client](https://github.com/elastic/kibana/pull/181926/files#diff-1d52c732de522ce954e4886516b0985854d8d9c5a31412dd2f6086591b940bd9R32), and 
* [ignore those fields during rule execution](https://github.com/elastic/kibana/pull/181926/files#diff-619f3684aae74ef2af0bd49dd3efb89650abef3106cf8d5c2c59d18a67f04f0fR160).

I believe this satisfies those requirements, but please let me know if we're missing something here.

Reply:
I think so - we're still learning how to do this ourselves :-)

The main thing to think about is rollbacks.  Let's call current version X, version with merged X+1 and "final" version X + 2.  

One of the primary problems is going to be alert param validation.  So if we're on version X + 1, and have to roll back to X, the new param will have to not be present, or the validation will fail for version X (since it doesn't know the parameter).  Of course, if you want to do something like test on version X + 1 with a feature flag on, for how it will work in X + 2, just realize the rule will fail with a validation check if we roll back to version X.